lib/identity/validation/validateIdentityType.js

```diff
@@ -0,0 +1,27 @@
+const Identity = require('../Identity');
+
+const ValidationResult = require('../../validation/ValidationResult');
+
+const UnknownIdentityTypeError = require('../../errors/UnknownIdentityTypeError');
+
+/**
+ * @typedef validateIdentityType
+ * @param {number} type
+ * @return ValidationResult
+ */
+function validateIdentityType(type) {
+  const result = new ValidationResult();
+
+  const isReservedType = type < Identity.MAX_RESERVED_TYPE;
+  const isRegisteredType = Object.values(Identity.TYPES).includes(type);
```

Comment:
There's a constant in the Identity class for this so we don't need to do this every time, `Identity.TYPES_ENUM`